test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -49,21 +123,99 @@
       (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
             context (tk-services/service-context jruby-service)
             pool-context (:pool-context context)]
-        (jruby-testutils/reduce-over-jrubies! pool-context 4 #(format "InstanceID = %s" %))
-        (is (= #{0 1 2 3}
-               (-> (jruby-testutils/reduce-over-jrubies! pool-context 4 (constantly "InstanceID"))
-                   set)))
-        (let [response (http-client/delete
-                         "https://localhost:8140/puppet-admin-api/v1/jruby-pool"
-                         ssl-request-options)]
-          (is (= 204 (:status response))))
-        ; wait until the flush is complete
-        (await (:pool-agent pool-context))
-        (is (every? true?
-                    (jruby-testutils/reduce-over-jrubies!
-                      pool-context
-                      4
-                      (constantly
-                        "begin; InstanceID; false; rescue NameError; true; end"))))))))
-
-;; TODO: test flush instance while pool flush is in progress
+        ;; set a ruby constant in each instance so that we can recognize them
+        (is (true? (set-constants-and-verify pool-context 4)))
+        (let [flush-complete (add-watch-for-flush-complete pool-context)]
+          (is (true? (trigger-flush)))
+          @flush-complete)
+        ;; now the pool is flushed, so the constants should be cleared
+        (is (true? (verify-no-constants pool-context 4)))))))
```

Comment:
Sorenson Principle violation!  

A principal runs a school, a principle is an idea you live by.